ext/standard/array.c

```diff
@@ -5986,15 +5986,15 @@ PHP_FUNCTION(array_reduce)
 		ZVAL_COPY(&args[1], operand);
 		fci.params = args;
 
+		zval_ptr_dtor(&result);
+
 		if (zend_call_function(&fci, &fci_cache) == SUCCESS && Z_TYPE(retval) != IS_UNDEF) {
 			zval_ptr_dtor(&args[1]);
 			zval_ptr_dtor(&args[0]);
```

Comment:
Both `if` and `else` branches have these `zval_ptr_dtor(&args[1]);` and `zval_ptr_dtor(&args[0]);` lines. Can't we move out of the check to avoid duplication?